### PR TITLE
[CN-111] Add remaining early years questions and hook up existing questions to flow

### PR DESCRIPTION
**Pull request by Lockyy**
[CNN-111](https://dfedigital.atlassian.net/browse/CN-111)

### Context

- Depends on https://github.com/DFE-Digital/npq-registration/pull/353
- Depends on https://github.com/DFE-Digital/npq-registration/pull/352

### Changes proposed in this pull request

- Adds Do you work in childcare?
- Adds Do you work in a nursery?
- Adds What kind of nursery do you work in?
- Hooks up existing early years questions to these new questions.
- Stores early years data on applications
- Adds early years data in registration wizard answers data

### Guidance to review

- Complete a journey through the public nursery flow in the app
- Complete a journey through the private nursery flow in the app

### CODE REVIEW

File: app/controllers/private_childcare_providers_controller.rb [hunk 1]
```diff
@@ -1,7 +1,6 @@
 class PrivateChildcareProvidersController < ApplicationController
   def index
     @private_childcare_providers = PrivateChildcareProvider
-      .open
```
Comment (Lockyy) [inline app/controllers/private_childcare_providers_controller.rb:4]: They're all open now.

File: app/lib/forms/choose_private_childcare_provider.rb [hunk 2]
```diff
@@ -4,9 +4,11 @@ class ChoosePrivateChildcareProvider < Base
 
     attr_accessor :institution_name, :institution_identifier
 
-    validates :institution_identifier, format: { with: /\APrivateChildcareProvider-\d+\z/, unless: -> { institution_identifier.blank? || institution_identifier == "other" } }
+    validates :institution_identifier, format: { with: /\APrivateChildcareProvider-.+\z/, unless: -> { institution_identifier.blank? || institution_identifier == "other" } }
```
Comment (Lockyy) [inline app/lib/forms/choose_private_childcare_provider.rb:7]: Ofsted URNs are often prefixed with EY. I've left the formatting open ended and instead validated the institution presence later on.

File: app/lib/forms/have_ofsted_urn.rb [hunk 3]
```diff
@@ -20,7 +20,11 @@ def next_step
     end
 
     def previous_step
-      :qualified_teacher_check
+      if wizard.store["works_in_nursery"] == "yes"
```
Comment (asmega) [inline app/lib/forms/have_ofsted_urn.rb:23]: one for next time, I added `Services::QueryStore` to try to contain the arbitrary store calls

File: app/models/private_childcare_provider.rb [hunk 4]
```diff
@@ -27,6 +27,10 @@ def urn
     provider_urn
```
Comment (asmega) [inline app/models/private_childcare_provider.rb:27]: i'd be tempted to store this under a new different field eg `ey_urn` or `childcare_urn` if they are actually a different from `urn`

**Pull request merged**